src/esp/assets/managers/AttributesManagerBase.h

```diff
@@ -1096,9 +1095,9 @@ AttributesManager<T>::removeTemplatesBySubstring(const std::string& subStr,
 }  // removeAllTemplates
 
 template <class T>
-typename AttributesManager<T>::AttribsPtr
-AttributesManager<T>::removeTemplateInternal(const std::string& templateHandle,
-                                             const std::string& sourceStr) {
+std::shared_ptr<T> AttributesManager<T>::removeTemplateInternal(
+    const std::string& templateHandle,
+    const std::string& sourceStr) {
```

Comment:
Does this work? In my opinion it's consistent without being excessively long, but other people have a strong dislike for trailing return types, so it depends.

```cpp
auto AttributesManager<T>::removeTemplateInternal(
    const std::string& templateHandle,
    const std::string& sourceStr) -> AttribsPtr {
```

Reply:
Oh, i like that.